compiler/qsc_parse/src/item.rs

```diff
@@ -33,7 +33,7 @@ pub enum Fragment {
 }
 
 pub(super) fn parse(s: &mut Scanner) -> Result<Box<Item>> {
-    let lo = s.peek().span.lo;
+    let t = s.peek();
     let doc = parse_doc(s);
     let attrs = many(s, parse_attr)?;
     let visibility = opt(s, parse_visibility)?;
```

Comment:
I know this is getting outside of the scope of the original fix, but we can hit the same recovery issue of floating doc comments if there is a floating attribute (like `@EntryPoint()`) or floating visibility (like `internal`) that is not followed by an item. Perhaps a way to address all of these would be to update `parse_doc` to return an `Option<String>` (and update the usage below in the success case to use `unwrap_or_else` to treat `None` and an empty string), then have logic that checks how far in processing an item we got before failing to find an actual item. If `visibility.is_some()` then we can indicate that an item is expected to follow visibility keyword, else if `!attrs.is_empty()` then we can indicate that an item is expected after one or more attributes, then finally if `doc.is_some()` indicate that an item is expected after a doc comment.

Reply:
I think this is a reasonable follow up issue. It would be much cleaner to use `Option` there, but it would require changing the `doc` field on types that can be documented. Unless you're okay with just providing an empty string slice, which I personally think looks a bit out of place.

Reply:
I sent ahead and updated the function signature of `parse_doc` and used `unwrap_or_default` where applicable. I can open a separate issue and open a follow up PR for this if that's okay?

Reply:
Thanks! Filing another issue for the floating attribute/visibility should be fine, I'd appreciate if you can add that and link to this comment.

Reply:
Ref #757